homeassistant/components/sensor/mychevy.py

```diff
@@ -0,0 +1,164 @@
+        return
+
+    hub = hass.data[DOMAIN]
+    sensors = [MyChevyStatus()]
+    for sconfig in SENSORS:
+        sensors.append(EVSensor(hub, sconfig))
+
+    add_devices(sensors)
+
+
+class MyChevyStatus(Entity):
+    """A string representing the charge mode."""
+
+    _name = "MyChevy Status"
+    _icon = "mdi:car-connected"
+
+    def __init__(self):
+        """Initialize sensor with car connection."""
+        self._state = None
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Register callbacks."""
+        self.hass.helpers.dispatcher.async_dispatcher_connect(
+            UPDATE_TOPIC, self.success)
+
+        self.hass.helpers.dispatcher.async_dispatcher_connect(
+            ERROR_TOPIC, self.error)
+
+    @callback
+    def success(self):
+        """Update state, trigger updates."""
+        if self._state != MYCHEVY_SUCCESS:
+            _LOGGER.info("Successfully connected to mychevy website")
+            self._state = MYCHEVY_SUCCESS
+        self.async_schedule_update_ha_state()
+
+    @callback
+    def error(self):
+        """Update state, trigger updates."""
+        if self._state != MYCHEVY_ERROR:
+            self.hass.components.persistent_notification.create(
+                "Error:<br/>Connection to mychevy website failed. "
+                "This probably means the mychevy to OnStar link is down.",
+                title=NOTIFICATION_TITLE,
+                notification_id=NOTIFICATION_ID)
+            self._state = MYCHEVY_ERROR
+        self.async_schedule_update_ha_state()
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return self._icon
+
+    @property
+    def name(self):
+        """Return the name."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state."""
+        return self._state
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return False
+
+
+class EVSensor(Entity):
+    """Base EVSensor class.
+
+    The only real difference between sensors is which units and what
+    attribute from the car object they are returning. All logic can be
+    built with just setting subclass attributes.
+
+    """
+
+    def __init__(self, connection, config):
+        """Initialize sensor with car connection."""
+        self._conn = connection
+        self._name = config.name
+        self._attr = config.attr
+        self._unit_of_measurement = config.unit_of_measurement
+        self._icon = config.icon
+        self._state = None
+
+        self.entity_id = ENTITY_ID_FORMAT.format(
+            '{}_{}'.format(DOMAIN, slugify(self._name)))
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Register callbacks."""
+        self.hass.helpers.dispatcher.async_dispatcher_connect(
+            UPDATE_TOPIC, self.async_update_callback)
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        if self._icon == "mdi:battery":
```

Comment:
It's a bit weird that you define a battery icon in the config and then overwrite that. I'd use the `_attr` as I'd to check icon for battery.